Mathlib/Algebra/Polynomial/Bivariate.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.RingTheory.AdjoinRoot
+
+/-!
+# Bivariate polynomials
+
+This file introduces the notation `R[X][Y]` for the polynomial ring `R[X][X]` in two variables,
+and the notation `Y` for the second variable, in the `Polynomial` scope.
+
+It also defines `Polynomial.evalEval` for the evaluation of a bivariate polynomial at a point
+on the affine plane, which is a ring homomorphism (`Polynomial.evalEvalRingHom`), as well as
+the abbreviation `CC` to view a constant in the base ring `R` as a bivariate polynomial.
+-/
+
+/-- The notation `Y` for `X` in the `Polynomial` scope. -/
+scoped[Polynomial] notation3:max "Y" => Polynomial.X (R := Polynomial _)
+
+/-- The notation `R[X][Y]` for `R[X][X]` in the `Polynomial` scope. -/
+scoped[Polynomial] notation3:max R "[X][Y]" => Polynomial (Polynomial R)
+
+namespace Polynomial
+
+noncomputable section
+
+variable {R S : Type*}
+
+/-- `evalEval x y p` is the evaluation `p(x,y)` of a two-variable polynomial `p : R[X][Y]`. -/
+abbrev evalEval [Semiring R] (x y : R) (p : R[X][Y]) : R := eval x (eval (C y) p)
```

Comment:
Can you add this somewhere?
```
lemma evalEval_C (p : R[X]) : (C p).evalEval x y = p.eval x := by
  rw [evalEval, eval_C]
```
Thanks!

Reply:
And since we have this, why don't we add this too:
```
lemma evalEval_X : X.evalEval x y = y := by
  rw [evalEval, eval_X, eval_C]
```
We can try to mimic the API for normal `eval` and see which are relevant to `evalEval`.